adapters/postgres/postgres.go

```diff
@@ -595,7 +596,36 @@ func (adapter *Postgres) CountByRequest(req *http.Request) (countQuery string, e
 	return
 }
 
-// Query process queries
+// Query process queries using the DB name from Context
+//
+// allows setting timeout
+func (adapter *Postgres) QueryCtx(ctx context.Context, SQL string, params ...interface{}) (sc adapters.Scanner) {
+	// use the db_name that was set on request to avoid runtime collisions
+	db, err := connection.GetFromPool(ctx.Value("db_name").(string))
```

Comment:
Before o use the value in context check if exists the value... 

If for some reason the return of `ctx.Value("db_name").(string)` is empty?